app/main.py

```diff
@@ -1,3 +1,18 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if args in cache_dict:
+            print("Getting from cache")
+            return cache_dict.get(args)
+        else:
+            result = func(*args)
+            cache_dict.update({args: result})
+            print("Calculating new result")
+            return result
```

Comment:
you return the same value for both cases so - use 1 return for them

Reply:
I don't understand what you mean, first return its a value from dict, and second its a value from function

Reply:
Yes, you add a second value to the dict, so you can retrieve it from it. Nataliia means it would be better to make one `return` for both cases and take it outside the if-else statement